custom-skills/OpenAiRedactionFunction.cs

```diff
@@ -0,0 +1,271 @@
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using OpenAI.Chat;
+
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using AISearch.CustomFunctions;
+
+
+namespace ChatCompletion;
+
+public class OpenAI_StructuredOutputs()
+{
+    
+    [Function("OpenAiRedactionFunction")]
+    public async Task<IActionResult> RedactSensitiveInfoWithOpenAI(
+            [HttpTrigger(AuthorizationLevel.Function, "get", "post")] HttpRequestData req)
+    {
+        string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+        OpenAiRedactionInputRecord inputRecord;
+
+        try
+        {
+            inputRecord = JsonConvert.DeserializeObject<OpenAiRedactionInputRecord>(requestBody);
+        }
+        catch (Exception ex)
+        {
+            return new BadRequestObjectResult($"Invalid input format: {ex.Message}");
+        }
+
+        if (string.IsNullOrEmpty(inputRecord?.Data?.Text))
+        {
+            return new BadRequestObjectResult("The input 'Text' field is required.");
+        }
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddAzureOpenAIChatCompletion(
+                deploymentName: Environment.GetEnvironmentVariable("OPENAI_DEPLOYMENT_NAME"),
+                endpoint: Environment.GetEnvironmentVariable("OPENAI_ENDPOINT"),
+                apiKey: Environment.GetEnvironmentVariable("OPENAI_API_KEY"))
+            .Build();
+
+        ChatResponseFormat chatResponseFormat = ChatResponseFormat.CreateJsonSchemaFormat(
+            jsonSchemaFormatName: "piiDetectionResult",
+            jsonSchema: BinaryData.FromString("""
+                {
+                    "type": "object",
+                    "properties": {
+                        "PiiFound": {
+                            "type": "boolean"
+                        },
+                        "PiiDetails": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "Text": { "type": "string" },
+                                    "Type": { "type": "string" },
+                                    "Context": { "type": "string" }
+                                },
+                                "required": ["Text", "Type", "Context"],
+                                "additionalProperties": false
+                            }
+                        }
+                    },
+                    "required": ["PiiFound", "PiiDetails"],
+                    "additionalProperties": false
+                }
+                """),
+            jsonSchemaIsStrict: true);
+
+        var executionSettings = new OpenAIPromptExecutionSettings
+        {
+            ResponseFormat = chatResponseFormat
+        };
+
+        var result = await kernel.InvokePromptAsync("""
```

Comment:
Please move the prompt to configuration so that a change in the prompt doesn't require a rebuild / republish